src/Azure.Functions.Sdk/ExtensionReference.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Mono.Cecil;
+
+namespace Azure.Functions.Sdk;
+
+/// <summary>
+/// Represents an Azure Functions extension reference.
+/// </summary>
+public static class ExtensionReference
+{
+    private const string ExtensionsInformationType = "Microsoft.Azure.Functions.Worker.Extensions.Abstractions.ExtensionInformationAttribute";
+
+    /// <summary>
+    /// Gets the root path of the Azure Functions SDK module.
+    /// </summary>
+    /// <param name="path">The path to the assembly.</param>
+    /// <param name="sourcePackageId">The source package ID.</param>
+    /// <param name="extensionReference">The resulting extension reference, if found.</param>
+    public static bool TryGetFromModule(
+        string path,
+        string sourcePackageId,
+        [NotNullWhen(true)] out ITaskItem? extensionReference)
+    {
+        // Read the assembly from the specified path
+        AssemblyDefinition assembly = AssemblyDefinition.ReadAssembly(path);
```

Comment:
Do we want to use the FunctionsAssembleyResolver?

Reply:
And does this need to dispose?
```suggestion
        using AssemblyDefinition assembly = AssemblyDefinition.ReadAssembly(path);
```